4-graceful-sigint/main.go

```diff
@@ -13,10 +13,38 @@
 
 package main
 
+import (
+	"context"
+	"os"
+	"os/signal"
+)
+
 func main() {
-	// Create a process
-	proc := MockProcess{}
+	ctx, cancel := context.WithCancel(context.Background())
```

Comment:
why introduce additional ctx if you could just listen on the notification channel itself?